sky/dag.py

```diff
@@ -14,6 +15,36 @@
 TaskOrName = Union['task.Task', str]
 
 
+@dataclasses.dataclass
+class TaskEdge:
+    """Represents an edge between two tasks in a DAG.
+
+    Attributes:
+        source: The upstream task.
+        target: The downstream task.
+        data_path: Path where data will be stored on the target node.
+        data_size_gb: Estimated size of the data in gigabytes.
+    """
+    source: 'task.Task'
+    target: 'task.Task'
+    data_path: Optional[str] = None
```

Comment:
source & target data path?

Reply:
If all data field is None, that should means there is no data between the tasks. only dependencies

also, should we have a new dataclass for the data field? so either all of three fields exists or none of them exists